src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs

```diff
@@ -90,9 +90,24 @@ public MBuild.Project Project
                 {
                     return Context.ProjectCollection.LoadProject(FileInfo.FullName);
                 }
-                catch (InvalidProjectFileException ex)
+                catch (InvalidProjectFileException)
                 {
-                    throw new UpgradeException(LocalizedStrings.InvalidProjectError, ex);
+                    // TODO Delete obj file retry - a change in platform version may cause load to fail for UWP/WinUI apps
+                    var fileDirectory = FileInfo.Directory;
+                    var dirToDelete = fileDirectory?.EnumerateDirectories().Where(directory => directory.Name == "obj").FirstOrDefault();
```

Comment:
Is this recursive such that it could unintentionally grab two `obj/` directories?
Also
```suggestion
                    var dirToDelete = fileDirectory?.EnumerateDirectories().FirstOrDefault(directory => directory.Name == "obj");
```